src/proxyClient.js

```diff
@@ -97,7 +97,14 @@ var frameId = 0;
 
 // Worker
 
-var worker = new Worker('{{{ filename }}}.js');
+var filename = '{{{ filename }}}.js';
+var workerURL = filename;
+var fileBytes = tryParseAsDataURI(filename);
+if (fileBytes) {
+  workerURL = URL.createObjectURL(new Blob([fileBytes], {type: 'application/javascript'}));
+}
+
+var worker = new Worker(workerURL);
```

Comment:
Are there any other places where you'd want similar changes? There's the various calls to `tryParseAsDataURI` in shell.js, which we could save a few bytes by removing, but the most obvious example I see is that two of the four functions in arrayUtils.js are currently only used for this option, so we could always split that into two files and only include the base64 half in jsifier.js when needed.

As far as how it would be implemented, would it be preferable to use `#if SINGLE_FILE || MEM_INIT_METHOD == 0` everywhere, or create a new separate setting value that gets toggled in emcc.py as needed?